src/lib/core/utils/wdk-filter-param-adapter.ts

```diff
@@ -286,8 +282,20 @@ export function makeHiddenVariablesInScope(
     });
   }
 
-  variablesByParentId[entity.id]?.forEach((variable) => {
-    _traverseDescendantVariables(variable, false);
+  const variableIds = new Set(entity.variables.map((variable) => variable.id));
```

Comment:
Note that this is duplicated in `entitiesToFields`. Is it worth deduplicating this operation?

Reply:
I think we should, eventually; but I think we will want to overhaul how we are handling variable trees in a larger context. You'll notice that we are doing a pseudo tree traversal of variables here, which means we have to re-implement the "parent logic" here.

An overhaul I've had in mind (but haven't had time to do, yet) is to construct the tree in the subsetting client, and to make that available as a part of the study metadata that the application code uses. I don't think there is an issue for this. I will make one in a bit and reference it here.

Reply:
https://github.com/VEuPathDB/web-eda/issues/1434